simtools/corsika/corsika_output.py

```diff
@@ -976,44 +1036,416 @@ def num_photons_per_event(self):
+            self._meta_dict["Title"] = function_dict["title"]
+            function = getattr(self, function_dict["function"])
+            hist_1D_list, x_edges_list = function()
+            x_edges_list = x_edges_list * function_dict["edges unit"]
+            if function_dict["function"] == "get_photon_density_distr":
+                histogram_value_unit = 1 / (function_dict["edges unit"] ** 2)
+            else:
+                histogram_value_unit = u.dimensionless_unscaled
+            hist_1D_list = hist_1D_list * histogram_value_unit
+            for i_histogram, _ in enumerate(x_edges_list):
+                if self.individual_telescopes:
+                    ecsv_file = (
+                        f"{function_dict['file name']}_"
+                        f"tel_index_{self.telescope_indices[i_histogram]}.ecsv"
+                    )
+                else:
+                    ecsv_file = f"{function_dict['file name']}_all_tels.ecsv"
+
+                table = self.fill_ecsv_table(
+                    hist_1D_list[i_histogram],
+                    x_edges_list[i_histogram],
+                    None,
+                    function_dict["edges"],
+                    None,
+                )
+                ecsv_file = Path(output_dir).joinpath(ecsv_file)
+                self._logger.info(f"Exporting histogram to {ecsv_file}.")
+                table.write(ecsv_file, format="ascii.ecsv", overwrite=True)
+
+    @property
+    def _dict_2D_distributions(self):
+        """
+        Dictionary to label the 2D distributions according to the class methods.
+
+        Returns
+        -------
+        dict:
+            The dictionary with information about the 2D distributions.
+        """
+        if self.__dict_2D_distributions is None:
+            self.__dict_2D_distributions = {
+                "counts": {
+                    "function": "get_2D_photon_direction_distr",
+                    "file name": "hist_2D_photon_direction_distr",
+                    "title": "Incoming directive cosines",
+                    "x edges": "x directive cosinus",
+                    "x edges unit": u.dimensionless_unscaled,
+                    "y edges": "y directive cosinus",
+                    "y edges unit": u.dimensionless_unscaled,
+                },
+                "density": {
+                    "function": "get_2D_photon_time_altitude",
+                    "file name": "hist_2D_photon_time_altitude_distr",
+                    "title": "Time of arrival vs altitude of emission",
+                    "x edges": "Time of arrival",
+                    "x edges unit": self.hist_config["hist_time_altitude"]["x axis"]["start"].unit,
+                    "y edges": "Altitude of emission",
+                    "y edges unit": self.hist_config["hist_time_altitude"]["y axis"]["start"].unit,
+                },
+                "direction": {
+                    "function": "get_2D_num_photons_distr",
```

Comment:
This should probably be "counts" instead of "direction"?

Reply:
Corrected